ruby/ql/lib/codeql/ruby/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -378,7 +379,9 @@ module Trees {
     override ControlFlowTree getChildElement(int i) { result = this.getArgument(i) }
   }
 
-  private class CaseTree extends PreOrderTree, CaseExpr, ASTInternal::TCaseExpr {
+  private class CaseTree extends PreOrderTree, CaseExpr {
+    CaseTree() { this instanceof CaseWhenExpr }
```

Comment:
Can you move this to the `extends` list?

Reply:
Unfortunately, that causes complaints about ambiguous method definitions.